cli-mirth/src/main/java/baltool/mirth/codegenerator/CodeGenerationUtils.java

```diff
@@ -0,0 +1,757 @@
+package baltool.mirth.codegenerator;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import io.ballerina.projects.BuildOptions;
+import io.ballerina.projects.DiagnosticResult;
+import io.ballerina.projects.ModuleDescriptor;
+import io.ballerina.projects.PackageCompilation;
+import io.ballerina.projects.directory.BuildProject;
+import io.ballerina.projects.util.ProjectUtils;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+import org.xml.sax.InputSource;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.StringReader;
+import java.util.stream.Collectors;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.ConnectException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import static baltool.mirth.Constants.BALLERINA_TOML_FILE;
+import static baltool.mirth.Constants.CONTENT;
+import static baltool.mirth.Constants.COPILOT_BACKEND_URL;
+import static baltool.mirth.Constants.DEV_COPILOT_BACKEND_URL;
+import static baltool.mirth.Constants.FILE_PATH;
+import static baltool.mirth.Constants.MAXIMUM_REPAIR_COUNT;
+import static baltool.mirth.codegenerator.HttpUtils.getHttpClient;
+import static baltool.mirth.codegenerator.HttpUtils.sendRequestAsync;
+import static baltool.mirth.codegenerator.HttpUtils.sendStreamRequestAsync;
+
+public class CodeGenerationUtils {
+
+    public static final boolean BALLERINA_DEV_UPDATE = Boolean.parseBoolean(
+            System.getenv("BALLERINA_DEV_UPDATE"));
+    private static final String TEMP_DIR_PREFIX = "logicapps-migration-tool-codegen-diagnostics-dir-";
+
+    private record GeneratedCode(Map<String, String> codeMap, JsonArray functions) {
+    }
+
+    public static String generateMirthChannelExecPlan(String channelXmlContent) {
```

Comment:
Why do we need this?

Reply:
removed in [82949cb](https://github.com/wso2/integration-bi-migration-assistant/pull/315/commits/82949cb50d9590e12c70a93f77e199fc9be3c835)